vm/qemu/qemu.go

```diff
@@ -534,7 +534,26 @@ func splitArgs(str, templateDir string, index int) (args []string) {
 		if arg == "" {
 			continue
 		}
-		arg = strings.ReplaceAll(arg, "{{INDEX}}", fmt.Sprint(index))
+		// 0-7 VFs/Device, if index >= 8, Device# += (index / 8)
+		if strings.Contains(arg, "{{INDEX}}") {
+			arg = strings.ReplaceAll(arg, "{{INDEX}}", fmt.Sprint(index%8))
```

Comment:
Wonder if this would work instead? Example arg: "vfio-pci,host=31:0a.{{INDEX%8}},addr=0x11"
`        if strings.Contains(arg, "{{INDEX%8}}") {
                if (index > 7) {
                        args1 := strings.Split(arg, "vfio-pci,host=")
                        args2 := strings.Split(args1[1], ":")
                        args3 := strings.Split(args2[1], ".")

                        devno, err := strconv.ParseInt(args3[0], 16, 64)
                        if err != nil {
                                return
                        }
                        devno += int64(index / 8)
                        str := ":" + args3[0]
                        fix := fmt.Sprintf(":%02x", devno)
                        arg = strings.ReplaceAll(arg, str, fix)
                }
                arg = strings.ReplaceAll(arg, "{{INDEX%8}}", fmt.Sprint(index%8))
        }
`

Reply:
Leaving {{INDEX}} as is and adding {{FN%8}} to handle the PCI BDF in this type of string:
"vfio-pci,host=31:0a.{{FN%8}},addr=0x11"

Latest commits are here: https://github.com/gkennedy12/syzkaller/tree/pull_072622_1